tests/e2e/README.md

```diff
@@ -1,117 +1,144 @@
+roles/container.admin (Kubernetes Engine Admin)
+Editor (on by default)
 
-`log_provider` and `project_id` must both be specified if one wishes to collect cluster logs.
+To set this, navigate to the IAM section of the Cloud Console and find your default GCE/GKE service account in the
+following form: projectNumber-compute@developer.gserviceaccount.com: by default it should just have the Editor role.
+Then in the Roles drop-down list for that account, find the Kubernetes Engine group and select the role Kubernetes
+Engine Admin. The Roles listing for your account will change to Multiple.
 
-### For all the following example, you always need to add:
-* `--auth_enable` if you want to include auth
-* `--cluster_wide` if you want to run the cluster wide installation and tests
-* `--use_initializer` if you want to do transparent sidecar injection
+## Step 2: Get cluster credentials
+```
+gcloud container clusters get-credentials ${CLUSTER_NAME} --zone ${ZONE} --project ${PROJECT_NAME}
+```
 
-### Example
-From the repo checkout root directory
+## Step 3: Create Clusterrolebinding
+```
+kubectl create clusterrolebinding prow-cluster-admin-binding    --clusterrole=cluster-admin    --user="your cluster user name(for example youralias@google.com)"
+```
 
-* Run tests with the latest stable version of istio according to istio.VERSION :
+## Step 4: Export test script variables
 
-`tests/e2e.sh --auth_enable`
+**Option 1:** Already committed changes to istio/istio master branch
+NOTE: SHA used for GIT_SHA is one that is already committed on istio/istio. You can pick any SHA you want.
+```
+export HUB="gcr.io/istio-testing"
+export GIT_SHA="d0142e1afe41c18917018e2fa85ab37254f7e0ca"
+```
 
-* Test commit in pilot repo, SHA:"dc738396fd21ab9779853635dd22693d9dd3f78a":
+**Option 2:** Testing local changes
 
-`tests/e2e.sh --pilot_hub=gcr.io/istio-testing --pilot_tag=dc738396fd21ab9779853635dd22693d9dd3f78a --istioctl_url=https://storage.googleapis.com/istio-artifacts/dc738396fd21ab9779853635dd22693d9dd3f78a/artifacts/istioctl --auth_enable`
+If you want to test on uncommitted changes to master istio:
+* Create a PR with your change.
+* This will trigger istio-presubmit.sh. At the end of this script, it creates docker images for mixer, pilot, ca, with
+your changes and upload them to container registry. See the logs of this istio-presubmit.sh and at the end there must
+be a SHA which you need to copy and set it as a GIT_SHA. Example from a log: the SHA to copy is marked in **bold**
 
-* If you want to run one specific test, you can do:
+	I1207 04:42:40.881] **0077bb73e0b9d2841f8c299f15305193e42dae0d**: digest: sha256:6f72528d475be56e8392bc3b833b94a815a1fbab8a70cd058b92982e61364021 size: 528
 
+* Then set the export variables again
 ```
-source istio.VERSION
-# Each time pilot SHA changes, get the matching istioctl binary
-# for instance on a mac:
-curl $ISTIOCTL_URL/istioctl-osx > ~/istioctl-osx ; chmod 755 ~/istioctl-osx
```

Comment:
replace maybe by "bazel build //pilot/cmd/istioctl" + add bazel-bin/... to the path ?

Reply:
https://github.com/istio/istio/issues/2036